tests/initrd.c

```diff
@@ -12,19 +12,9 @@ static void dump_file(const char *path) {
 
   char buffer[1000];
   memset(buffer, '\0', sizeof(buffer));
-  uio_t uio;
-  iovec_t iov;
-  uio.uio_op = UIO_READ;
-
-  /* Read entire file - even too much. */
-  uio.uio_iovcnt = 1;
-  uio.uio_vmspace = get_kernel_vm_map();
-  uio.uio_iov = &iov;
-  uio.uio_offset = 0;
-  iov.iov_base = buffer;
-  iov.iov_len = sizeof(buffer);
-  uio.uio_resid = sizeof(buffer);
 
+  uio_single_t uio;
+  make_uio_kernel(&uio, UIO_READ, buffer, sizeof(buffer), 0);
```

Comment:
I'll just describe in more details why I don't like this extension.

Most of people to join this project aren't going to be C experts. 
The following line:
```
    res = VOP_READ(v, &uio); 
```

Is very confusing. Someone unfamiliar with these extensions will probably think there is some magic cast going on in here. I'm 100% confident that simply naming the field in `uio_single_t` (say `uio`) and making:

``` 
make_uio_kernel(&uio.uio, UIO_READ, buffer, sizeof(buffer), 0); 
```

Is more readable.